pandas/io/json/_json.py

```diff
@@ -923,19 +954,39 @@ def read(self) -> DataFrame | Series:
         """
         obj: DataFrame | Series
         with self:
-            if self.lines:
-                if self.chunksize:
-                    obj = concat(self)
-                elif self.nrows:
-                    lines = list(islice(self.data, self.nrows))
-                    lines_json = self._combine_lines(lines)
-                    obj = self._get_object_parser(lines_json)
+            if self.engine == "pyarrow":
+                pyarrow_json = import_optional_dependency("pyarrow.json")
+                pa_table = pyarrow_json.read_json(self.data)
+                if (
+                    self.use_nullable_dtypes
+                    and get_option("mode.dtype_backend") == "pyarrow"
+                ):
+                    from pandas.arrays import ArrowExtensionArray
+
+                    obj = DataFrame(
+                        {
+                            col_name: ArrowExtensionArray(pa_col)
+                            for col_name, pa_col in zip(
+                                pa_table.column_names, pa_table.itercolumns()
+                            )
+                        }
+                    )
+                    return obj
+                obj = pa_table.to_pandas()
```

Comment:
Can we convert to our nullables here as well if specified? Should go out of the box with the types mapper we use for parquet?